twine-2-jsonoutput-doc.md

```diff
@@ -2,11 +2,11 @@
 
 This specification defines a series of properties for encoding story and passage data compatible with Twine 2 and related tools using [JavaScript Object Notation (JSON)](https://www.json.org/json-en.html).
 
-## Story Encoding
+## Story Data Encoding
 
-To maintain close compatibility with the existing formats of [Twine 2 HTML](https://github.com/iftechfoundation/twine-specs/blob/master/twine-2-htmloutput-spec.md) and [Twee 3 notation](https://github.com/iftechfoundation/twine-specs/blob/master/twee-3-specification.md), story properties encoded in JSON mirror those found in the [`StoryData` passage in Twee 3 notation](https://github.com/iftechfoundation/twine-specs/blob/master/twee-3-specification.md#storydata):
+To maintain close compatibility with the existing output formats of [Twine 2 HTML](https://github.com/iftechfoundation/twine-specs/blob/master/twine-2-htmloutput-spec.md) and [Twee 3 notation](https://github.com/iftechfoundation/twine-specs/blob/master/twee-3-specification.md), story properties encoded in JSON mirror those found in the [`StoryData` passage in Twee 3 notation](https://github.com/iftechfoundation/twine-specs/blob/master/twee-3-specification.md#storydata):
 
-- `ifid`: (string) Required. Maps to `<tw-storydata ifid>`.
+- `ifid`: (string) Optional. Maps to `<tw-storydata ifid>`.
```

Comment:
I'm curious, why is the `ifid` **optional** in this JSON related specification when it is **required** in both the [twee 3](https://github.com/iftechfoundation/twine-specs/blob/master/twee-3-specification.md#storydata) and [twine 2 html output](https://github.com/iftechfoundation/twine-specs/blob/master/twine-2-htmloutput-spec.md#story-data) specifications?

Reply:
> Both Twine 2 HTML and Twee 3 generally represent whole stories. This new paragraph explains how some passages containing content, CSS, or JavaScript code could be encoded and then combined together with other JSON content as "story fragments" to make one single story during compilation. It's my hope this opens the door to plugins and CSS templates stored as JSON.

To allow for partial story encodings. This is a known problem with both Twine 2 HTML and Twee 3 code and examples. Potentially, this encoding could represent parts or some of a story rather than inherit the IFID same problem.